llvm/lib/Target/AIE/AIELegalizerHelper.cpp

```diff
@@ -152,28 +161,31 @@ bool AIELegalizerHelper::legalizeG_BUILD_VECTOR(LegalizerHelper &Helper,
   MIRBuilder.buildUndef(Src);
 
   const AIEBaseInstrInfo *II = ST.getInstrInfo();
-  MachineOperand *OperandBegin = MI.operands_begin(),
-                 *Operand = MI.operands_end() - 1;
-  while (Operand != OperandBegin) {
-    Register Reg = Operand->getReg();
+  MachineOperand *OperandBegin = MI.operands_begin() + 1,
+                 *Operand = MI.operands_begin() + 1,
+                 *OperandEnd = MI.operands_end();
+  while (Operand != OperandEnd) {
+    Register EltReg = Operand->getReg();
+    LLT EltRegTy = MRI.getType(EltReg);
     Register Dst = MRI.createGenericVirtualRegister(VecTy);
 
-    if (DstVecSize == 512 && Operand == OperandBegin + 1)
+    if (DstVecSize == 512 && Operand == OperandBegin) {
       Dst = DstReg;
```

Comment:
nit: we've created an unused virtual register.

Reply:
Is this about the register `Dst`, we have used later.